File: exam/EXAM_MATERIAL/window_std.m
function S = window_std(I)
    window = ones(5)/25;
    K_I2 = imfilter(I.^2, window, 'symmetric');
    KI_2 = imfilter(I, window, 'symmetric').^2;
    S = sqrt(K_I2 - KI_2); 
end

File: exam/EXAM_MATERIAL/window_std.py
import numpy as np
import scipy.ndimage

def window_std(I):
    window = np.ones(shape=(5, 5))/25
    K_I2 = scipy.ndimage.convolve(I**2, window, mode='reflect')
    KI_2 = scipy.ndimage.convolve(I, window, mode='reflect')**2
    return np.sqrt(K_I2 - KI_2)